kong/pdk/log.lua

```diff
@@ -559,6 +560,9 @@ do
     end
 
     local host_port = ctx.host_port or var.server_port
+    
+    local tries = (ctx.balancer_data or {}).tries or {}
+    setmetatable(tries, cjson.array_mt)
```

Comment:
here you are creating an empty table if there is none. imo we should not do that, it adds more processing and data to the logs without conveying additional information. The value being absent should do.

Reply:
Also setting a metatable on an existing table (which is a shared data structure) doesn't seem like a good idea. I'd rather attach the metatable right when the `tries` table is being created, somewhere in the `balancer` phase iirc.
That would make it more clear for every user of that table, it is an array, and hence be easier on future us for maintenance.

Reply:
You are right, this is not a good place to do these things that should been handled at balancer phase.

Reply:
Now the `ctx.balancer_data` may contains empty tries table or tries array, which is uncertained. I will open an issue to describe this problem and close this pr.